processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -534,79 +535,6 @@ public void fillBitmapsFromUnsortedEncodedArray(
     }
   }
 
-  @Override
-  public ValueMatcher makeIndexingValueMatcher(
-      final String matchValue,
-      final IncrementalIndexStorageAdapter.EntryHolder holder,
-      final int dimIndex
-  )
-  {
-    final int encodedVal = getEncodedValue(matchValue, false);
-    final boolean matchOnNull = Strings.isNullOrEmpty(matchValue);
-    if (encodedVal < 0 && !matchOnNull) {
-      return new BooleanValueMatcher(false);
-    }
-
-    return new ValueMatcher()
```

Comment:
Yes.  You will want to do this before starting the timer on the benchmark and also make sure you run each of the options with enough iterations to cause optimizations to actually occur as well as the backing out of those optimizations.  This is generally black magic unless you are running a debug build of the jvm where you can turn on logging of optimizations (something I've read about doing, but never done myself).